// components/Statuses/Statuses.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Statuses.css';

const cx = classNames.bind(styles);

export default class Statuses extends Component {

  static get propTypes() {
    return {
      text: PropTypes.string,
      New: PropTypes.boolean,
      Open: PropTypes.boolean,
      Resolved: PropTypes.boolean,
      Pending: PropTypes.boolean,
      Escalated: PropTypes.boolean,
      Closed: PropTypes.boolean,
      Trashed: PropTypes.boolean,
      Reversed: PropTypes.boolean,
      // status: string
    };
  };

  // renderText() {
  //   const { status } = this.props;
  //   if (!status) {
  //     return null;
  //   } else {
  //     return "dwadwada";
  //   // return <FormattedMessage id={`${status}`} defaultMessage="Default value if translation is not available" />
  //   // return `${status}` ;
  //  };

  render() {
    const { New, Open, Resolved, Pending, Escalated, Closed, Trashed, Reversed } = this.props;

    const className = cx(styles.Statuses, {
      New,
      Open,
      Resolved,
      Pending, 
      Escalated, 
      Closed, 
      Trashed, 
      Reversed
    });

    return (
      <div className={className}>
        {/* {this.renderText()} */}
      </div>
    );
  }
}
