// components/Tag/Tag.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Tag.css';

const cx = classNames.bind(styles);

export default class Tag extends Component {

  static get propTypes() {
    return {
      text: PropTypes.string,
      Pink: PropTypes.string,
      Yellow: PropTypes.string,
      Green: PropTypes.string,
      Blue: PropTypes.string,
      LightBlue: PropTypes.string,
      Purple: PropTypes.string,
      Turquoise: PropTypes.string,
      DarkPurple: PropTypes.string,
      Red: PropTypes.string,
      Grey: PropTypes.string
    };
  };

  render() {
    const { Pink, Yellow, Green, Blue, LightBlue, Purple, DarkPurple, Turquoise, Red, Grey } = this.props;

    const className = cx(styles.Tag, {
      Pink,
      Yellow,
      Green,
      Blue,
      LightBlue,
      Purple,
      Turquoise,
      DarkPurple,
      Red,
      Grey
    });

    return (
      <div className={className}>
        {this.props.text}
      </div>
    );
  }
}
