// components/Tag/Tag.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Tag.css';

const cx = classNames.bind(styles);

export default class Tag extends Component {

  static get propTypes() {
    return {
      text: PropTypes.string,
      Pink: PropTypes.boolean,
      Yellow: PropTypes.boolean,
      Green: PropTypes.boolean,
      Blue: PropTypes.boolean,
      LightBlue: PropTypes.boolean,
      Purple: PropTypes.boolean,
      Turquoise: PropTypes.boolean,
      DarkPurple: PropTypes.boolean,
      Red: PropTypes.boolean,
      Grey: PropTypes.boolean
    };
  };

  render() {
    const { Pink, Yellow, Green, Blue, LightBlue, Purple, DarkPurple, Turquoise, Red, Grey } = this.props;

    const className = cx(styles.Tag, {
      Pink,
      Yellow,
      Green,
      Blue,
      LightBlue,
      Purple,
      Turquoise,
      DarkPurple,
      Red,
      Grey
    });

    return (
      <div className={className}>
        {this.props.text}
      </div>
    );
  }
}
