// components/Link/Link.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Link.css';

const cx = classNames.bind(styles);

export default class Link extends Component {

  static get propTypes() {
    return {
      text: PropTypes.string,
      Icon: PropTypes.string,
      Blue: PropTypes.string,
      White: PropTypes.string,
      Green: PropTypes.string,
      Red: PropTypes.string,
      Black: PropTypes.string
    };
  };

  renderIcon() {
    if (this.props.Icon) {
      return(
        <div className={this.props.Icon}></div>
      );
    }
    return null;
  }

  render() {
    const { Blue, White, Green, Red, Black } = this.props;

    const className = cx(styles.Link, {
      Blue,
      White,
      Green,
      Red,
      Black
    });

    return (
      <a className={className}>
          {this.renderIcon()}
          <div className={styles.Text}>
            {this.props.text}
          </div>
      </a>
    );
  }
}
