// components/Dropdown/Dropdown.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Dropdown.css';

const cx = classNames.bind(styles);

export default class Dropdown extends Component {

  static get propTypes() {
    return {
      children: PropTypes.Array,
      styleName: PropTypes.string 
    };
  };

  handleClick() {
    console.log('handleclick');
    if (!this.state.popupVisible) {
      
      // attach/remove event handler
      document.addEventListener('click', this.handleOutsideClick, false);
    } else {
      document.removeEventListener('click', this.handleOutsideClick, false);
    }

    this.setState(prevState => ({
       popupVisible: !prevState.popupVisible,
    }));
  }
  
  handleOutsideClick(e) {
    console.log('outside', e.target, this.refs.test);
    // ignore clicks on the component itself
    if (this.refs.test.contains(e.target)) {
      return;
    }

    this.handleClick();
  }

  render() {
    const { styleName } = this.props;

    const className = cx(styles.Dropdown, styleName);

    return (
      <div className={className} onClick={this.handleClick} ref="test">
        {this.props.children}
      </div>
    );
  }
}
