// components/ChatBot/Plugin/Element/Element.jsx
import React from 'react'
import Image from './Image'
import Button from './Button'
import Heading, { SUBTITLE_HEADING_LEVEL, TITLE_HEADING_LEVEL } from './Heading'
import Text from './Text'
import QuickReply from './QuickReply'
import ComposedElement from './ComposedElement'

export const types = {
  BUTTON: 'BUTTON',
  HEADING: 'HEADING',
  TITLE: 'TITLE',
  SUBTITLE: 'SUBTITLE',
  TEXT: 'TEXT',
  IMAGE: 'IMAGE',
  QUICK_REPLY: 'QUICK_REPLY'
}

export default function Element ({ id, type, text, url, postback, elements, onClick }) {
  const headingTypes = [
    types.HEADING,
    types.TITLE,
    types.SUBTITLE
  ]

  if (type === types.BUTTON) {
    return <Button label={text} onClick={onClick} postback={postback} />
  }
  if (headingTypes.indexOf(type) > -1) {
    const level = type === types.SUBTITLE ? SUBTITLE_HEADING_LEVEL : TITLE_HEADING_LEVEL
    return <Heading content={text} level={level}/>
  }
  if (type === types.TEXT) {
    return <Text content={text} />
  }
  if (type === types.IMAGE) {
    return <Image src={url} />
  }
  if (type === types.QUICK_REPLY) {
    return <QuickReply elements={elements} onSelect={onClick} />
  }
  if (Array.isArray(elements)) {
    return <ComposedElement elements={elements} onClick={onClick} />
  }
  return null
}
