// components/ChatBot/Plugin/index.js
import React, { Component } from 'react'
import ComposedElement from './Element/ComposedElement'
import styles from './index.css'
import classname from 'classnames'

class Plugin extends Component {

  scroller = null
  requestAnimationFrameId = null

  SCROLL_DIRECTION_RIGHT = 1
  SCROLL_DIRECTION_LEFT = -1

  scrollTo = (direction) => {
    if (this.scroller !== null) {
      this.scroller.scrollLeft += (direction * 10)
    }
    this.requestAnimationFrameId = requestAnimationFrame(() => this.scrollTo(direction))
  }

  stopScroll = () => {
    cancelAnimationFrame(this.requestAnimationFrameId)
  }

  startScroll = (direction) => {
    if (this.requestAnimationFrameId !== null) {
      cancelAnimationFrame(this.requestAnimationFrameId)
    }
    this.requestAnimationFrameId = requestAnimationFrame(() => this.scrollTo(direction))
  }

  handleLeftArrowClick = () => {
    this.startScroll(this.SCROLL_DIRECTION_LEFT)
  }

  handleRightArrowClick = () => {
    this.startScroll(this.SCROLL_DIRECTION_RIGHT)
  }

  render () {
    const { elements, onClick } = this.props;
    const isItCarusel = elements.length > 1;
    const caruselClass = classname({ [styles.Carusel]: isItCarusel });

    return (<div className={styles.PluginFrameWrapper}>
      <div className={styles.ArrowWrapper}>
        {isItCarusel && <div className={styles.LeftRow} onMouseLeave={this.stopScroll} onMouseUp={this.stopScroll} onMouseDown={this.handleLeftArrowClick} />}
      </div>
      <div className={styles.PluginFrame} ref={(el) => this.scroller = el}>
        <div className={caruselClass}>
          {elements.map(element => (<div class={styles.Plugin}>
            <ComposedElement elements={element.elements} onClick={onClick} />
          </div>))}
        </div>
      </div>
      <div className={styles.ArrowWrapper}>
        {isItCarusel && <div className={styles.RightRow} onMouseLeave={this.stopScroll} onMouseUp={this.stopScroll} onMouseDown={this.handleRightArrowClick} />}
      </div>
    </div>);
  }
}

Plugin.prototype.defaultProps = {
  elements: []
}

export default Plugin
