// components/ChatBot/Plugin/index.js
import React, { Component } from 'react'
import ComposedElement from './Element/ComposedElement'
import Element from './Element'
import styles from './index.css'
import classname from 'classnames'

class Plugin extends Component {

  static defaultProps = {
    elements: []
  }

  scroller = null
  requestAnimationFrameId = null

  SCROLL_DIRECTION_RIGHT = 1
  SCROLL_DIRECTION_LEFT = -1

  scrollTo = (direction) => {
    if (this.scroller !== null) {
      this.scroller.scrollLeft += (direction * 10)
    }
    this.requestAnimationFrameId = requestAnimationFrame(() => this.scrollTo(direction))
  }

  stopScroll = () => {
    cancelAnimationFrame(this.requestAnimationFrameId)
  }

  startScroll = (direction) => {
    if (this.requestAnimationFrameId !== null) {
      cancelAnimationFrame(this.requestAnimationFrameId)
    }
    this.requestAnimationFrameId = requestAnimationFrame(() => this.scrollTo(direction))
  }

  handleLeftArrowClick = () => {
    this.startScroll(this.SCROLL_DIRECTION_LEFT)
  }

  handleRightArrowClick = () => {
    this.startScroll(this.SCROLL_DIRECTION_RIGHT)
  }

  renderCarusel(elements, onClick) {
    return (
      <div className={styles.PluginFrameWrapper}>
        <div className={styles.ArrowWrapper}>
          {<div className={styles.LeftRow} onMouseLeave={this.stopScroll} onMouseUp={this.stopScroll} onMouseDown={this.handleLeftArrowClick} />}
        </div>
        <div className={styles.PluginFrame} ref={(el) => this.scroller = el}>
          <div className={styles.Carusel}>
            {elements.map(element => (<div className={styles.Plugin}>
              <ComposedElement elements={element.elements} onClick={onClick} />
            </div>))}
          </div>
        </div>
        <div className={styles.ArrowWrapper}>
          {<div className={styles.RightRow} onMouseLeave={this.stopScroll} onMouseUp={this.stopScroll} onMouseDown={this.handleRightArrowClick} />}
        </div>
      </div>
    );
  }

  renderPlugin(elements, onClick) {
    return (
      <div className={styles.PluginFrameWrapper}>
        <div className={styles.PluginFrame}>
          <div className={styles.Plugin}>
            <ComposedElement elements={elements} onClick={onClick} />
          </div>
        </div>
      </div>
    );
  }

  render () {
    const { elements, onClick } = this.props;
    return elements.length > 1 ? this.renderCarusel(elements, onClick) : this.renderPlugin(elements, onClick)
  }
}

export default Plugin
