// components/Channel/Channel.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Channel.css';

const cx = classNames.bind(styles);

export default class Picture extends Component {

  static get propTypes() {
    return {
      styleName: PropTypes.string,
      Facebook: PropTypes.string, 
      Twitter: PropTypes.string, 
      Instagram: PropTypes.string, 
      Google: PropTypes.string, 
      Youtube: PropTypes.string, 
      Linkedin: PropTypes.string, 
      Livechat: PropTypes.string, 
      Email: PropTypes.string, 
      Forum: PropTypes.string, 
      Vkontakte: PropTypes.string, 
      Listening: PropTypes.string, 
      Size60: PropTypes.string, 
      Size40: PropTypes.string, 
      Size20: PropTypes.string, 
      Size16: PropTypes.string, 
      DM: PropTypes.string 
    };
  };

  render() {
    const { styleName, Facebook, Twitter, Instagram, Google, Youtube, Linkedin, Livechat, Email, Forum, Vkontakte, Listening, Size60, Size40, Size20, Size16, DM } = this.props;

    const className = cx(styles.Channel, styleName, {
      Facebook, 
      Twitter, 
      Instagram, 
      Google, 
      Youtube, 
      Linkedin, 
      Livechat, 
      Email, 
      Forum, 
      Vkontakte, 
      Listening, 
      Size60, 
      Size40, 
      Size20, 
      Size16, 
      DM
    });

    return (
      <div className={className}>
      </div>
    );
  }
}
