// components/Textarea/Textarea.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Textarea.css';

const cx = classNames.bind(styles);

export default class Textarea extends Component {

  static get propTypes() {
    return {
      children: PropTypes.Array,
      name: PropTypes.string,
      rows: PropTypes.number,
      value: PropTypes.string,
      placeholder: PropTypes.string,
      isDisabled: PropTypes.string,
      isError: PropTypes.string,
      onChange: PropTypes.function,
      desc: PropTypes.string
    };
  };

  render() {
    const { isDisabled, isError } = this.props;

    const className = cx(styles.Textarea, {
      Disabled: isDisabled,
      Error: isError
    });

    return (
      <div className={className}>
        <div className={styles.Label}>
          {this.props.label}                
        </div>
        <div className={styles.Field}>
        <textarea 
          name={this.props.name} 
          rows={this.props.rows}
          value={this.props.value} 
          placeholder={this.props.placeholder} 
          disabled={this.props.isDisabled} 
          onChange={this.props.onChange} />
        </div>
        <div className={`${styles.Desc} ${styles.isRed} `}>
          {this.props.desc}                
        </div>
      </div>
    );
  }
}
