// components/Button/Button.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Button.css';

const cx = classNames.bind(styles);

export default class Button extends Component{

  static get propTypes() {
    return {
      onClick: PropTypes.Function,
      text: PropTypes.string,
      Icon: PropTypes.string,
      Disabled: PropTypes.string,
      Reversed: PropTypes.string,
      Negative: PropTypes.string,
      Cancel: PropTypes.string,
      Small: PropTypes.string
    };
  };
  
  renderIcon() {
    if (this.props.Icon) {
      return (
        <div className={this.props.Icon}></div>
      );
    }

    return null;
  }

  render() {
    const { isDisabled, isReversed, isNegative, isCancel, Small, Wide } = this.props;

    const className = cx(styles.Button, {
      Disabled: isDisabled,
      Reversed: isReversed,
      Negative: isNegative,
      Cancel: isCancel,
      Small,
      Wide 
    });

    return (
      <button className={className} onClick={this.props.onClick}>
          {this.renderIcon()}
          {this.props.text} 
      </button>
    );
  }
}
