// components/Button/Button.js

// @flow

import React, { Component } from 'react';
import classNames from 'classnames/bind';
import styles from './Button.css';
import PropTypes from 'prop-types';

// type Props = {
//   onClick: Function,
//   text: string,
//   Icon: string,
//   Disabled: string,
//   Reversed: string,
//   Negative: string,
//   Cancel: string,
//   Small: string
// }

const cx = classNames.bind(styles);

export default class Button extends Component{

  static propTypes = {
    onClick: PropTypes.Function,
    text: PropTypes.string,
    Icon: PropTypes.string,
    Disabled: PropTypes.string,
    Reversed: PropTypes.string,
    Negative: PropTypes.string,
    Cancel: PropTypes.string,
    Small: PropTypes.string
  };
  
  renderIcon() {
    if (this.props.Icon) {
      return (
        <div className={this.props.Icon}></div>
      );
    }

    return null;
  }

  render() {
    const { isDisabled, isReversed, isNegative, isCancel, Small, Wide } = this.props;

    const className = cx(styles.Button, {
      Disabled: isDisabled,
      Reversed: isReversed,
      Negative: isNegative,
      Cancel: isCancel,
      Small,
      Wide 
    });

    return (
      <button className={className} onClick={this.props.onClick}>
          {this.renderIcon()}
          {this.props.text} 
      </button>
    );
  }
}
